File: csource/native/getime.c
#include <time.h>

typedef struct
    {
    double cputim;
    } chrono_t;

extern chrono_t chrono_;

double
getime(void)
    {
    return clock()/(double)CLOCKS_PER_SEC-chrono_.cputim;
    }

int
getime_(double* elapsed)
    {
    *elapsed = getime();
    return 0;
    }

File: classic/getime.f
c
c
c     ###################################################
c     ##  COPYRIGHT (C)  1990  by  Jay William Ponder  ##
c     ##              All Rights Reserved              ##
c     ###################################################
c
c     ##############################################################
c     ##                                                          ##
c     ##  subroutine getime  --  get elapsed CPU time in seconds  ##
c     ##                                                          ##
c     ##############################################################
c
c
c     "getime" gets elapsed CPU time in seconds for an interval
c
c
      subroutine getime (elapsed)
      implicit none
      include 'chrono.i'
      real*8 elapsed
c
c
c     elapsed time for the interval is the current total CPU
c     time minus the total time at the start of the interval
c
      call clock (elapsed)
      elapsed = elapsed - cputim
      return
      end
